plugins/dynamodb-positions-storage/src/main/java/com/github/bsideup/liiklus/dynamodb/DynamoDBPositionsStorage.java

```diff
@@ -51,57 +43,27 @@
 
     public static final String RANGE_KEY_FIELD = "groupId";
 
-    AmazonDynamoDBAsync dynamoDB;
+    DynamoDbAsyncClient dynamoDB;
 
     String tableName;
 
     @Override
     public Publisher<Positions> findAll() {
-        var request = new ScanRequest(tableName);
-
-        AtomicBoolean done = new AtomicBoolean(false);
-
-        return Mono
-                .<ScanResult>create(sink -> dynamoDB.scanAsync(request, new DefaultAsyncHandler<>(sink)))
-                .<List<Map<String, AttributeValue>>>handle(((scanResult, sink) -> {
-                    try {
-                        if (scanResult.getCount() <= 0) {
-                            done.set(true);
-                            sink.complete();
-                        } else {
-                            if (scanResult.getLastEvaluatedKey() == null || scanResult.getLastEvaluatedKey().isEmpty()) {
-                                done.set(true);
-                            }
-                            sink.next(scanResult.getItems());
-                        }
-                    } catch (Exception e) {
-                        sink.error(e);
-                    }
-                }))
-                .repeat(() -> !done.get())
-                .flatMapIterable(it -> it)
+        return Flux.defer(() -> Flux.from(dynamoDB.scanPaginator(req -> req.tableName(tableName))))
+                .flatMapIterable(ScanResponse::items)
                 .map(item -> new Positions(
-                        item.get("topic").getS(),
-                        GroupId.ofString(item.get("groupId").getS()),
+                        item.get("topic").s(),
+                        GroupId.ofString(item.get("groupId").s()),
                         toPositions(item)
                 ));
     }
 
     @Override
     public CompletionStage<Map<Integer, Long>> findAll(String topic, GroupId groupId) {
-        var request = new GetItemRequest()
-                .withTableName(tableName)
-                .withConsistentRead(true)
-                .withKey(toKey(topic, groupId));
-
-        return Mono
-                .<GetItemResult>create(sink -> {
-                    var future = dynamoDB.getItemAsync(request, new DefaultAsyncHandler<>(sink));
-                    sink.onCancel(() -> future.cancel(true));
-                })
+        return Mono.defer(() -> Mono.fromCompletionStage(dynamoDB.getItem(req -> req.tableName(tableName).consistentRead(true).key(toKey(topic, groupId)))))
```

Comment:
Could you please check Reactor's version?
There is `Mono.fromCompletionStage(Supplier<CompletionStage>)` to avoid `Mono.defer` now

Reply:
Also, this line is tooooo long, would make sense to split the request builder line by line